src/index.js

```diff
@@ -173,20 +177,25 @@ const getAttribute = (attributes, name) => {
   return getAttributeValues(attributes, path)[0];
 };
 
+const getCompareValue = (condition, attributes) => {
+  if ('target' in condition) {
+    return getAttribute(attributes, condition.target);
+  }
+  return condition.value;
+};
+
 /**
- * @param condition a condition to evaluate against the given
- * value.
- * @param condition.comparison a comparison operation string from the available
- * list of comparison operations.
- * @param condition.value the value to be compared against the function's
- * "value" argument using the condition's operation.
- * @param value the value to be compared against the condition's value
- * using the condition's comparison operator.
- *
- * @returns `true` if the comparision matches, `false` if there is a mismatch.
+ * @returns `true` if the comparision matches, `false` if there is a mismatch,
+ *           and `undefined` if the target value is not known to compute the
+ *           result.
  */
-const compare = (condition, value) => {
-  const compareValue = condition.value;
+const compare = (condition, value, attributes) => {
```

Comment:
should we move a lot of these functions to object style instead already?
```suggestion
const compare = ({ condition, value, attributes }) => {
```

Reply:
> should we move a lot of these functions to object style instead already?
> 
> ```suggestion
> 
> const compare = ({ condition, value, attributes }) => {
> 
> ```

Probably so. There's a number of improvements to be made in this repo, starting with TS. This func was here before the prev commit, so I wanted to keep it the same way for now.